pkg/kubectl/apply/strategy/replace_visitor.go

```diff
@@ -65,7 +74,9 @@ func (v replaceStrategy) MergeEmpty(e apply.EmptyElement) (apply.Result, error)
 // replace returns the local value if specified, otherwise it returns the remote value
 // this works regardless of the approach
 func (v replaceStrategy) doReplace(e apply.Element) (apply.Result, error) {
-	// TODO: Check for conflicts
+	if v.options.FailOnConflict {
```

Comment:
We need to have very good documentation for what constitutes a conflict in `FailOnConflict `.  How does it behave for lists, maps, primitives?